lib/util/errors.js

```diff
@@ -0,0 +1,123 @@
+"use strict";
+
+const { stripHash } = require("./url");
+
+const GenericError = exports.GenericError = class GenericError extends Error {
+  constructor (message, source) {
+    super();
+
+    this.message = message;
+    this.source = source;
+    this.path = [];
+  }
```

Comment:
All the errors that are thrown by this library should be [Ono](https://github.com/JS-DevTools/ono) errors.  Ono enhances the errors to work better for logging and serialization, and some other libraries depend on this enhanced functionality.

I just published a [new version of Ono](https://github.com/JS-DevTools/ono/blob/master/CHANGELOG.md) (v7.1.0) that adds an `Ono.extend()` method, which you can call in the constructors of each of these error classes.  i.e. `Ono.extend(this)`